### PR TITLE
Fixes #21, #23, #24, and #25.

### LINKED ISSUE #21: Enhancement: allow adding multiple folders to the list in the AutoFiler tab
After the user clicked on the Add... button, multiple selections should be allowed in the "Filer: Open" window.

### LINKED ISSUE #24: AutoFiler tab, Add button behaves like Edit button
After a monitored folder in AutoFiler tab is edited, no more folders can be added. The Add button from now on will replace the edited entry regardless whether it is selected.

### PR DESCRIPTION
**Pull request by owenca**
#21 - the Add button can add multiple folders now
 #23 - sort the folders like the way Tracker does
 #25 - remove the Edit button. this also fixes #24

### CODE REVIEW

**Comment by humdingerb** [comment]
Looks good, but I have a few suggestions:
* When clicking the Add button, the opened file panel should be labeled "AutoFiler: Add folders". The default button should be "Add" instead of "Open".
* When clicking the Edit button, the opened file panel should be labeled "AutoFiler: Edit folder". The default button should be "Choose" instead of "Open".
* The file panel for the Edit button allows for multi-selection. That's good, because maybe I'd like to replace that one folder with several. But when clicking "Open", only one folder is replacing the old folder; the others should be added as well.

**Comment by owenca** [comment]
I will take care of the first two bullets. Regarding the third one, I only enabled multi-selection for the Add panel. The Edit panel didn't change. (The previous versions also have this weird problem. Sometimes you can select multiple entries in the Edit panel, but Shift+Up/Down and Ctrl-click won't work.) So leave the Edit panel single/multi-selection as is?

**Comment by humdingerb** [comment]
If you want to, you can make the Edit panel also deal with multiple selections correctly, but I don't think it's really essential, just a little convenience. In any case, it seems to be worthy a bug report at Haiku's Trac. Brilliant would be of course if you can find the bug and attach a patch there right away. :)

Since you have an fAddPanel, you may want to call the other one "fEditPanel" instead of the generic fFilePanel.

**Comment by owenca** [comment]
> - When clicking the Add button, the opened file panel should be labeled "AutoFiler: Add folders". The default button should be "Add" instead of "Open".
> - When clicking the Edit button, the opened file panel should be labeled "AutoFiler: Edit folder". The default button should be "Choose" instead of "Open".
> 
> Since you have an fAddPanel, you may want to call the other one "fEditPanel" instead of the generic fFilePanel.

Done.

> In any case, it seems to be worthy a bug report at Haiku's Trac. Brilliant would be of course if you can find the bug and attach a patch there right away. :)

I will investigate it and try to find out why.

**Comment by owenca** [comment]
> - The file panel for the Edit button allows for multi-selection. That's good, because maybe I'd like to replace that one folder with several. But when clicking "Open", only one folder is replacing the old folder; the others should be added as well.

> If you want to, you can make the Edit panel also deal with multiple selections correctly, but I don't think it's really essential, just a little convenience.

It will be tricky if we allow multiple-selection for Edit because the list of selected folders may contain the folder that is being edited and other duplicate folders. I will wait until after #23 is implemented.

**Comment by humdingerb** [comment]
OK. Then this PR can be merged?

**Comment by owenca** [comment]
I have cleaned it up a little and think it's good to go now.

File: sources/AutoFilerTab.cpp [hunk 1]
```diff
@@ -216,7 +223,9 @@ AutoFilerTab::MessageReceived(BMessage* msg)
 		}
 		case MSG_SHOW_ADD_PANEL:
 		{
-			fFilePanel->Show();
+			fAddPanel->Window()->SetTitle("AutoFiler: Add folders");
+			fAddPanel->SetButtonLabel(B_DEFAULT_BUTTON, "Add");
```
Comment (humdingerb) [inline sources/AutoFilerTab.cpp]: The SetTitle() and SetButtonLabel() could be done once right after creating the file panels, right?

File: sources/AutoFilerTab.cpp [hunk 2]
```diff
@@ -226,12 +235,14 @@ AutoFilerTab::MessageReceived(BMessage* msg)
...
-			fFilePanel->SetPanelDirectory(item->Text());
+			fEditPanel->SetPanelDirectory(item->Text());
 
 			BMessage panelMsg(MSG_FOLDER_CHOSEN);
 			panelMsg.AddInt32("index", selection);
-			fFilePanel->SetMessage(&panelMsg);
-			fFilePanel->Show();
+			fEditPanel->SetMessage(&panelMsg);
+			fEditPanel->Window()->SetTitle("AutoFiler: Edit folder");
+			fEditPanel->SetButtonLabel(B_DEFAULT_BUTTON, "Choose");
```
Comment (humdingerb) [inline sources/AutoFilerTab.cpp]: The SetTitle() and SetButtonLabel() could be done once right after creating the file panels, right?

Comment (owenca) [inline sources/AutoFilerTab.cpp]: I had all of them in the constructor at first, but it caused one panel's buttons to interfere with the other's and vice versa, probably due to bugs in BFilePanel. Now that we are going to remove the Edit button from the AutoFiler tab, the problem will no longer exist. I will move them back to the constructor when removing the Edit button.

Comment (owenca) [inline sources/AutoFilerTab.cpp]: Done.

**Comment by owenca** [comment]
14bbd1b fixes #21, #23, #24, and #25.

**Comment by humdingerb** [comment]
WRT #23 Alphabetically sort the monitored AutoFiler folders, the sorting order differs to how Tracker is doing it. The capital letter folders are sorted in before the lower case folders. Tracker does the sorting case-insensitive. You see a way to fix that?

**Comment by owenca** [comment]
Done. Please let me know if you find any problems.

**Comment by humdingerb** [comment]
Works perfectly, thanks!

**Pull request merged**